model/Ssd1306.cpp

```diff
@@ -0,0 +1,276 @@
+        }
+
+
+
+void Ssd1306::Pixel(int x, int y, bool set) 
+{
+	  if (x >= 0 && x < 128 && y >= 0 && y < 64) {
+		if (set) {
+			ScreenBuffer[(x + (y / 8) * 128) + 1] |= (1 << (y % 8)); 
+		}
+         else 
+        {
+		   ScreenBuffer[(x + (y / 8) * 128) + 1] &= (~(1 << (y % 8)));
+		}
+    
+	  }
+}
+
+void Ssd1306::SetPixel(int x, int y) 
+{
+    Pixel(x, y, true);
+}
+
+void Ssd1306::drawBitmap(const uint8_t *bitmap, uint8_t x, uint8_t y, uint8_t width, uint8_t height) {
+ 
+   for(int16_t xd=0; xd<width; xd++) {
+        for(int16_t yd=0; yd<height; yd++ ) {
+            Pixel(x+xd, y+yd, bitmap[width * yd + xd]==1);
+        }
+    }
+  
+}
+
+
+
+
+void Ssd1306::DrawCircle(int x0, int y0, int radius)
+ {
+  int x = 0, y = radius;
+	int dp = 1 - radius;
+	do {
+		if (dp < 0)
+			dp = dp + 2 * (++x) + 3;
+		else
+			dp = dp + 2 * (++x) - 2 * (--y) + 5;
+
+		SetPixel(x0 + x, y0 + y);     //For the 8 octants
+		SetPixel(x0 - x, y0 + y);
+		SetPixel(x0 + x, y0 - y);
+		SetPixel(x0 - x, y0 - y);
+		SetPixel(x0 + y, y0 + x);
+		SetPixel(x0 - y, y0 + x);
+		SetPixel(x0 + y, y0 - x);
+		SetPixel(x0 - y, y0 - x);
+
+	} while (x < y);
+
+  SetPixel(x0 + radius, y0);
+  SetPixel(x0, y0 + radius);
+  SetPixel(x0 - radius, y0);
+  SetPixel(x0, y0 - radius);
+}
+
+void Ssd1306::DrawRectangle(int x, int y, int width, int height) 
+   {
+     if (width < 0) return;
+     if (height < 0) return;
+
+     for (int i = x; i < x + width; i++) 
+     {
+       SetPixel(i, y);
+       SetPixel(i, y + height - 1);
+     }
+
+     for (int i = y; i < y + height; i++)
+     {
+       SetPixel(x, i);
+       SetPixel(x + width - 1, i);
+     }            
+       
+   }
+
+void Ssd1306::DrawFilledRectangle(int x, int y, int width, int height)
+ {
+            for (int lx = x; lx < width + x; lx++)
+                for (int ly = y; ly < height + y; ly++)
+                    SetPixel(lx, ly);
+ }
+
+void Ssd1306::DrawLine(int x1, int y1, int x2, int y2)
+{
+    int y,x,b,direction=1;
+    float m;
+
+    if (x1<0 && x2<0) return;
+    if (x1>127 && x2>127) return;
+    if (y1<0 && y2<0) return;
+    if (y1>63 && y2>63) return;
+
+    if ((x1 == x2) && (y1 == y2)) {
+        SetPixel(x1,y1);
+        return;
+    }
+
+    if (abs(x2-x1)<abs(y2-y1)) {
+        if (y2<y1) direction =-1;
+        m=((float)x2-(float)x1)/((float)y2-(float)y1);
+        b=x1-m*y1+0.5;
+        y=y1-direction;
+        while(y!=y2) {
+            y+=direction;
+            x=m*y+b+0.5;
+            SetPixel(x,y);
+        }
+    } else {
+        if (x2<x1) direction =-1;
+        m=((float)y2-(float)y1)/((float)x2-(float)x1);
+        b=y1-m*x1+0.5;
+        x=x1-direction;
+        while(x!=x2) {
+            x+=direction;
+            y=m*x+b+0.5;
+            SetPixel(x,y);
+        }
+    }
+
+}
+
+void Ssd1306::DrawSymbol(int x, int y, char letter, int HScale, int VScale) 
+{
+     int index = 5 * (letter - 32);
+
+         for (int h = 0; h < 5; h++) {
```

Comment:
We use `int` here but types like `uint8_t` in other places, we should be consistent.